services/api-server/tests/unit/test__code_syntax.py

```diff
@@ -1,12 +1,102 @@
-# pylint:disable=unused-variable
-# pylint:disable=unused-argument
-# pylint:disable=redefined-outer-name
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
 
 
+from uuid import UUID
+
+import pytest
+from faker import Faker
+from fastapi import APIRouter, FastAPI, status
+from fastapi.testclient import TestClient
 from pytest_simcore.helpers.utils_pylint import (
     assert_no_pdb_in_code,
     assert_pylint_is_passing,
 )
+from simcore_service_api_server.models.schemas.solvers import (
+    Solver,
+    SolverKeyId,
+    VersionStr,
+)
+
+
+@pytest.fixture
+def client() -> TestClient:
+    # overrides client.
+    # WARNING: this is NOT httpx.AsyncClient
+
+    router = APIRouter(prefix="/solvers")
+
+    @router.get("/{solver_key:path}/releases/{version}/jobs/{job_id:uuid}")
+    async def get_job(
+        solver_key: SolverKeyId,
+        version: VersionStr,
+        job_id: UUID,
+    ):
+        return {
+            "action": get_job.__name__,
+            "solver_key": solver_key,
+            "version": version,
+            "job_id": job_id,
+        }
+
+    @router.post("/{solver_key:path}/releases/{version}/jobs/{job_id:uuid}:start")
+    async def start_job(
+        solver_key: SolverKeyId,
+        version: VersionStr,
+        job_id: UUID,
+    ):
+        return {
+            "action": start_job.__name__,
+            "solver_key": solver_key,
+            "version": version,
+            "job_id": job_id,
+        }
+
+    @router.post("/{solver_key:path}/releases/{version}/jobs/{job_id:uuid}:stop")
+    async def stop_job(
+        solver_key: SolverKeyId,
+        version: VersionStr,
+        job_id: UUID,
+    ):
+        return {
+            "action": stop_job.__name__,
+            "solver_key": solver_key,
+            "version": version,
+            "job_id": job_id,
+        }
+
+    the_app = FastAPI()
+    the_app.include_router(router, prefix="/v0")
+
+    return TestClient(the_app)
+
+
+def test_fastapi_route_name_parsing(client: TestClient, faker: Faker):
+    #
+    # This is a fast test for compatibility.
+    # Will help in early detection of issues when fastapi library is upgraded
+    #
```

Comment:
I guess you had a problem with this, that's why you added this test. Maybe also detail the reason for the test in this comment.

Reply:
Yes, there seems to be a difference in newer versions with the way paths are parsed. It was caught at the integration tests and at that point it was too late and did not know the exactly library that produced this. By adding this check I can catch it early enough.
I think the test code is self explanatory and there is no need for extra comments .. i can add the fragment above nonetheless.